h2o-core/src/main/java/water/util/Log.java

```diff
@@ -22,261 +23,234 @@
-    String res = sb.toString();
-    if( H2O.SELF_ADDRESS == null ) { // Oops, need to buffer until we can do a proper header
-      INIT_MSGS.add(res);
-      return;
-    }
-    if( INIT_MSGS != null ) {   // Ahh, dump any initial buffering
-      String host = H2O.SELF_ADDRESS.getHostAddress();
-      _preHeader = fixedLength(host + ":" + H2O.API_PORT + " ", 22) + fixedLength(H2O.PID + " ", 6);
-      ArrayList<String> bufmsgs = INIT_MSGS;  INIT_MSGS = null;
-      if (bufmsgs != null) for( String s : bufmsgs ) write0(INFO, true, s);
-    }
-    write0(lvl, stdout, res);
-  }
-
-  private static void write0( int lvl, boolean stdout, String s ) {
-    StringBuilder sb = new StringBuilder();
-    String hdr = header(lvl);   // Common header for all lines
-    write0(sb, hdr, s);
-
-    // stdout first - in case log4j dies failing to init or write something
-    if(stdout && !_quiet) System.out.println(sb);
-
-    // log something here
-    org.apache.log4j.Logger l4j = _logger != null ? _logger : createLog4j();
-    switch( lvl ) {
-    case FATAL:l4j.fatal(sb); break;
-    case ERRR: l4j.error(sb); break;
-    case WARN: l4j.warn (sb); break;
-    case INFO: l4j.info (sb); break;
-    case DEBUG:l4j.debug(sb); break;
-    case TRACE:l4j.trace(sb); break;
-    default:
-      l4j.error("Invalid log level requested");
-      l4j.error(s);
-    }
+  public static void ignore(Throwable e) {
+    ignore(e,"[h2o] Problem ignored: ");
   }
 
-  private static void write0( StringBuilder sb, String hdr, String s ) {
-    if( s.contains("\n") ) {
-      for( String s2 : s.split("\n") ) { write0(sb,hdr,s2); sb.append("\n"); }
-      sb.setLength(sb.length()-1);
-    } else {
-      sb.append(hdr).append(s);
-    }
+  public static void ignore(Throwable e, String msg) {
+    ignore(e, msg, true);
   }
 
-  // Build a header for all lines in a single message
-  private static String header( int lvl ) {
-    String nowString = Timer.nowAsLogString();
-    String s = nowString +" "+_preHeader+" "+
-      fixedLength(Thread.currentThread().getName() + " ", 10)+
-      LVLS[lvl]+": ";
-    return s;
+  public static void ignore(Throwable e, String msg, boolean printException) {
+    debug(msg + (printException? e.toString() : ""));
   }
 
-  // A little bit of startup buffering
-  private static ArrayList<String> INIT_MSGS = new ArrayList<>();
+  public static void setQuiet(boolean q) { quietLogging = q; }
+  public static boolean getQuiet() { return quietLogging; }
 
   public static void flushStdout() {
-    if (INIT_MSGS != null) {
-      for (String s : INIT_MSGS) {
+    if (initialMsgs != null) {
+      for (String s : initialMsgs) {
         System.out.println(s);
       }
-
-      INIT_MSGS.clear();
+      initialMsgs.clear();
     }
   }
 
-  /**
-   * @return This is what should be used when doing Download All Logs.
-   */
+  /** Get directory where the logs are stored */
   public static String getLogDir() throws Exception {
```

Comment:
Does not make sense to throw untyped exception, in this case we can simply throws RuntimeException and remove `trows` from method signature.